test/src/test/java/org/corfudb/runtime/view/ManagementViewTest.java

```diff
@@ -1516,4 +1517,104 @@ public void queryClusterStatus() throws Exception {
+                .addSequencer(SERVERS.PORT_0)
+                .addSequencer(SERVERS.PORT_1)
+                .buildSegment()
+                .buildStripe()
+                .addLogUnit(SERVERS.PORT_0)
+                .addLogUnit(SERVERS.PORT_1)
+                .addToSegment()
+                .addToLayout()
+                .setClusterId(UUID.randomUUID())
+                .build();
+        bootstrapAllServers(l);
+        corfuRuntime = getRuntime(l).connect();
+
+        CorfuRuntime managementRuntime0 = getManagementServer(SERVERS.PORT_0)
+                .getManagementAgent().getCorfuRuntime();
+        CorfuRuntime managementRuntime1 = getManagementServer(SERVERS.PORT_1)
+                .getManagementAgent().getCorfuRuntime();
+
+        // Waiting for management servers to send the bootstrap sequencer request and be ready
+        // to detect failures to speed up test time.
+        CFUtils.within(
+                CompletableFuture.allOf(
+                        getManagementServer(SERVERS.PORT_0).getManagementAgent().getRecoveryBarrierFuture(),
+                        getManagementServer(SERVERS.PORT_1).getManagementAgent().getRecoveryBarrierFuture()),
+                PARAMETERS.TIMEOUT_NORMAL
+        ).join();
+
+        IStreamView streamView = corfuRuntime.getStreamsView()
+                .get(CorfuRuntime.getStreamID("testStream"));
+        final byte[] payload = "test".getBytes();
+        final int num = 10;
+        for (int i = 0; i < num; i++) {
+            streamView.append(payload);
+        }
+
+        // Setting aggressive timeouts
+        setAggressiveTimeouts(l, corfuRuntime, managementRuntime0, managementRuntime1);
+
+        setAggressiveDetectorTimeouts(SERVERS.PORT_0, SERVERS.PORT_1);
+
+        final Semaphore semaphore = new Semaphore(2);
+        semaphore.acquire(2);
+
+        //Release the semaphore on invocation.
+        Runnable systemDownHandler = () -> {
+            semaphore.release();
+            throw new UnreachableClusterException("Runtime stalled. aborting operation.");
+        };
+        managementRuntime0.registerSystemDownHandler(systemDownHandler);
+        managementRuntime1.registerSystemDownHandler(systemDownHandler);
+        final int sysDownTriggerLimit = 3;
+        managementRuntime0.getParameters().setSystemDownHandlerTriggerLimit(sysDownTriggerLimit);
+        managementRuntime1.getParameters().setSystemDownHandlerTriggerLimit(sysDownTriggerLimit);
+
+
+        TestRule testRule = new TestRule()
+                .matches(m -> m.getMsgType().equals(CorfuMsgType.SET_EPOCH))
+                .drop();
+        addClientRule(managementRuntime0, testRule);
+        addClientRule(managementRuntime1, testRule);
+
+        addServerRule(SERVERS.PORT_1, new TestRule().matches(m -> m.getMsgType().equals(CorfuMsgType.READ_RESPONSE)).drop());
+
+        Layout newLayout = new Layout(l);
+        newLayout.setEpoch(newLayout.getEpoch() + 1);
```

Comment:
This doesn;t really test how the UnreachableClusterException is handled, which is the code path that your changing. I think we need a better test.